authz/api/http/endpoint.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/mainflux/authz"
+)
+
+func addPolicy(svc authz.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(addPolicyReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+
+		policy := authz.Policy{
+			Action:  req.Action,
+			Subject: req.Subject,
+			Object:  req.Object,
+		}
+
+		added, err := svc.AddPolicy(ctx, req.token, policy)
+		if err != nil {
+			return addPolicyRes{created: false}, err
+		}
+
+		return addPolicyRes{created: added}, nil
```

Comment:
Is this `added` flag needed? Can we simplify to 
```
		if err := svc.AddPolicy(ctx, req.token, policy); err != nil {
			return addPolicyRes{created: false}, err
		}

		return addPolicyRes{created: true}, nil

Reply:
`added` means added, if it is already there, then `added` is false, so it is additonal info that is useful I think

Reply:
Maybe you can check if err.contains(ErrDuplicatedPolicy) and return `return addPolicyRes{created: false}, nil`

Reply:
@mteodor Or maybe simply return `ErrConflict` as we are doing for User registration. Otherwise it will be unclear that there is a conflict.

Reply:
users and policies are different type of entities, and `casbin` itself doesnt return an error when you add already existing policy so I followed that principle